suite/tests/api/ir_riscv64.h

```diff
@@ -0,0 +1,100 @@
+/* **********************************************************
+ * Copyright (c) 2024 Institute of Software Chinese Academy of Sciences (ISCAS).
+ * All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of ISCAS nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL ISCAS OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+static byte buf[8192];
```

Comment:
This is problematic in a .h file: I think this needs to be moved to a new .c file; since we need that for this, I would move the function definitions below there as well and leave just declarations here.

Alternatively, remove this buf definition and have the (inline) functions below take in the buffer as a parameter.

Reply:
> This is problematic in a .h file: I think this needs to be moved to a new .c file; since we need that for this, I would move the function definitions below there as well and leave just declarations here.

IMHO, most of the problems here (duplicated symbols, include guard) aren't really problems. Our tests are simple-structured thus it's easy to keep a header not being included several times. There's already usage for aarch64.

Didn't read the CMakeLists.txt line by line, but keeping each test contained in exact one corresponding .c should simplify the script.

If you think these problems do matter, I'm willing to split out a new .c for sure.

> 
> Alternatively, remove this buf definition and have the (inline) functions below take in the buffer as a parameter.